File: backend/model/Turf.js
const mongoose = require("mongoose");
const { Schema } = mongoose;

// Turf Schema
const TurfSchema = new Schema({
  size: {
    type: String, // Changed from array to a single string
    required: true,
  },
  admin: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "admin",
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
  location: {
    type: String, // Simplified location from nested schema to a single string
    required: true,
  },
  openTime: {
    type: String,
    required: true,
  },
  closeTime: {
    type: String,
    required: true,
  },
  price: {
    type: Number, // Renamed "rate" to "price" for consistency
    required: true,
  },
  slotDuration: {
    type: Number,
    required: true,
  },
  images: [String],
  timeStamp: {
    type: Date,
    required: true,
    default: Date.now,
  },
});

const Turf = mongoose.model("turf", TurfSchema);

module.exports = Turf;

File: frontend/lib/models/turf.dart
class Turf {
  final String id;
  final String name;
  final String size;
  final String location;
  final String admin;
  final String openTime;
  final String closeTime;
  final double price;
  final int slotDuration;
  final List<String> image; // Array of Base64 images

  Turf({
    required this.id,
    required this.name,
    required this.size,
    required this.location,
    required this.admin,
    required this.openTime,
    required this.closeTime,
    required this.price,
    required this.slotDuration,
    required this.image, // Array of Base64 images
  });

  factory Turf.fromJson(Map<String, dynamic> json) {
    return Turf(
      id: json['_id'],
      name: json['name'],
      size: json['size'],
      location: json['location'],
      admin: json['admin'],
      openTime: json['openTime'],
      closeTime: json['closeTime'],
      price: double.parse(json['price'].toString()),
      slotDuration: int.parse(json['slotDuration'].toString()),
      image: List<String>.from(
          json['images']), // Parse images as an array of strings
    );
  }
}
